python-client/giskard/scanner/performance.py

```diff
@@ -16,36 +16,80 @@
 from ..slicing.category_slicer import CategorySlicer
 from ..slicing.multiscale_slicer import MultiscaleSlicer
 
+from giskard.ml_worker.testing.tests.performance import test_diff_f1, test_diff_rmse, test_diff_accuracy, \
+    test_diff_recall, test_diff_precision
+
 
 class PerformanceScan:
     tags = ["performance", "classification", "regression"]
 
-    def __init__(self, model: Optional[BaseModel] = None, dataset: Optional[Dataset] = None):
+    def __init__(self, model: Optional[BaseModel] = None, dataset: Optional[Dataset] = None, test_names=None):
         self.params = {}
         self.model = model
         self.dataset = dataset
+        self.test_names = test_names
 
-    def run(self, model: Optional[BaseModel] = None, dataset: Optional[Dataset] = None, slicer="opt"):
+    def run(self, model: Optional[BaseModel] = None, dataset: Optional[Dataset] = None, slicer="opt", test_names=None,
+            threshold=0.1):
         model = model or self.model
         dataset = dataset or self.dataset
+        test_names = test_names or self.test_names
+
+        if test_names == "f1":
+            test = test_diff_f1
+        if test_names == "accuracy":
+            test = test_diff_accuracy
+        if test_names == "recall":
+            test = test_diff_recall
+        if test_names == "precision":
+            test = test_diff_precision
+        if test_names == "rmse":
+            test = test_diff_rmse
+        if test_names is None:
+            if model.is_classification:
+                test = test_diff_f1
+            else:
+                test = test_diff_rmse
 
         if model is None:
             raise ValueError("You need to provide a model to test.")
         if dataset is None:
             raise ValueError("You need to provide an evaluation dataset.")
 
-        # Should check model type
         # …
 
         # Calculate loss
         meta = self._calculate_meta(model, dataset)
         slices = self.find_slices(dataset.select_columns(model.meta.feature_names), meta, slicer)
-
-        # @TODO: filter this @mathieuroques ;)
-        issues = [Issue(slice_fn, model, dataset) for slice_fn in slices]
+        issues = self._find_issues(slices, model, dataset, test, threshold)
 
         return PerformanceScanResult(issues)
 
+    def _find_issues(self, slices, model, dataset, test, threshold):
+        issues = []
+        for s in slices:
+            actual_slices_size, reference_slices_size, metric, passed = self._diff_test(s, test, threshold)
+            if not passed:
+                issues.append(Issue(s, metric, actual_slices_size, reference_slices_size, model, dataset))
+
+        return issues
+
+    def _diff_test(self, data_slice, test, threshold):
+        # Convert slice to Giskard Dataframe
+        slice_dataset = Dataset(data_slice.data)
```

Comment:
```py
sliced_dataset = self.dataset.slice(slice_fn)
```

Reply:
`self.model` and `self.dataset` might be undefined, you should pass explicitly to `_diff_test`